python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -570,6 +575,69 @@ def with_resources(self, resource_defs: Mapping[str, ResourceDefinition]) -> "As
             group_names_by_key=self.group_names_by_key,
         )
 
+    def without_resources(self) -> "AssetsDefinition":
+        # In the case where we provided a system-created resource key, we need
+        # to remove it without requiring users provide it later.
+        if isinstance(self.node_def, OpDefinition):
+
+            output_defs = []
+            for output_def in self.op.output_defs:
+                asset_key = self.keys_by_output_name.get(output_def.name)
+                if output_def.io_manager_key == io_manager_key_for_asset_key(asset_key):
+                    io_manager_key = DEFAULT_IO_MANAGER_KEY
+                    new_required_keys = {
+                        io_manager_key,
+                        *{
+                            key
+                            for key in self.op.required_resource_keys
+                            if key != io_manager_key_for_asset_key(self.asset_key)
+                        },
+                    }
+
+                else:
+                    io_manager_key = output_def.io_manager_key
+                    new_required_keys = self.op.required_resource_keys
```

Comment:
this seems like the wrong place to put this logic, as new_requried_keys is dependent on whichever output definition is the last in the list. I don't think required_resource_keys will contain the keys of io managers in the general case, so it's probably fine to omit this part.